src/infrastructure/validation/logger/ConsoleLogger.java

```diff
@@ -0,0 +1,219 @@
+	 */
+	private boolean inTestMode = false;
+	
+	/**
+	 *  Creates an object that logs to the console, if enabled and available
+	 */
+	protected ConsoleLogger(List<LogLevel> enabledLogLevelsList, boolean enableTestMode) {
+		
+		timeStampFormat = DateTimeFormatter.ofPattern("yyyy:MM:dd HH:mm:ss");
+		
+		console = System.console();
+		
+		// set testMode to enabled/disabled
+		// variable enableTestMode is a boolean which is a primitive data type and,
+		// has a default value of false if unassigned
+		inTestMode = enableTestMode;
+		
+		if(null == console) {
+			
+			// no console to print found
+			// disable all the log levels i.e. have no messages logged
+			enableErrorLog = false;
+			enableWarningLog = false;
+			enableSuccessLog = false;
+			enableInfoLog = false;
+		
+		} else if (null == enabledLogLevelsList) {
+		
+			// no info specified
+			// use default value
+			enableErrorLog = false;
+			enableWarningLog = false;
+			enableSuccessLog = false;
+			enableInfoLog = false;
+		
+		} else {
+			
+			// check the List to see which LogLevel enum are sent
+			// for every member of the List, set the corresponding boolean to true i.e. enabled
+			if (enabledLogLevelsList.contains(LogLevel.ERROR)) {
+				enableErrorLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.WARNING)) {
+				enableWarningLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.SUCCESS)) {
+				enableSuccessLog = true;
+			}
+			if (enabledLogLevelsList.contains(LogLevel.INFO)) {
+				enableInfoLog = true;
+			}
+			
+		}
+		
+	}
+
+	@Override
+	synchronized public void log(ModuleID moduleIdentifier, LogLevel level, String message) {
+		
+		LocalDateTime now = LocalDateTime.now();
+		String formatDateTime = now.format(timeStampFormat);
+
+		String logTimeStamp = "";
+
+		String logModulePart = "";
+		
+		String logLevelPart = "";
+		
+		if(!inTestMode) {
+
+			logTimeStamp = "["+formatDateTime+"]";
+			
+			logModulePart = "["+moduleIdentifier.toString()+"]";
+			
+			logLevelPart = "["+level.toString()+"]";
+	
+		} else {
+			
+			/**
+			 *  logger is in testMode
+			 *  as per infrastructure lead and test harness requirement,
+			 *  disable logs from all other modules
+			 */			
+			switch(moduleIdentifier) {
+			case TEST:
+				break;
+			default:
+				return;
+			}
+		}
+		
+		String logMessage = logTimeStamp+logModulePart+logLevelPart+message;
+
+		printToConsole(level, logMessage);
+	}
+
+	/**
+	 *  private helper method that can print the log messages to Console,
+	 *  if the corresponding log level is enabled
+	 *  
+	 * @param level LogLevel enum that specifies the kind of log message
+	 * @param message String that holds the value to be printed to console
+	 */
+	private void printToConsole(LogLevel level, String message) {
+		
+		String ANSI_COLOR = getColorFromLevel(level);
+		
+		switch(level) {
+		case ERROR:
+			if(enableErrorLog) {
+				System.out.println(ANSI_COLOR+message+ANSI_RESET);
```

Comment:
Data member `console` should be used instead of `System.out.println`. `console.printf()` works I guess.
Lines 172, 177, 182, 187.

Reply:
Adding on to what @wildcat26 said, there can be sync issues in using two different stream objects. Please rectify this.

Reply:
Sorry, there was confusion. This is fine. Marking as resolved.